src/m365/connection/commands/connection-remove.spec.ts

```diff
@@ -0,0 +1,157 @@
+import assert from 'assert';
+import sinon from 'sinon';
+import auth, { AuthType, CertificateType, CloudType } from '../../../Auth.js';
+import { Logger } from '../../../cli/Logger.js';
+import { telemetry } from '../../../telemetry.js';
+import { pid } from '../../../utils/pid.js';
+import { session } from '../../../utils/session.js';
+import commands from '../commands.js';
+import command from './connection-remove.js';
+import { CommandInfo } from '../../../cli/CommandInfo.js';
+import { settingsNames } from '../../../settingsNames.js';
+import { sinonUtil } from '../../../utils/sinonUtil.js';
+import { CommandError } from '../../../Command.js';
+import { cli } from '../../../cli/cli.js';
+
+describe(commands.REMOVE, () => {
+  let log: string[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+  let commandInfo: CommandInfo;
+
+  before(() => {
+    sinon.stub(auth, 'clearConnectionInfo').resolves();
+    sinon.stub(auth, 'storeConnectionInfo').resolves();
+    sinon.stub(auth, 'removeConnectionInfo').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    commandInfo = cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+
+    sinon.stub(auth, 'ensureAccessToken').resolves();
+
+    auth.connection.active = true;
```

Comment:
I don't think so, as it will remove connections and change the active connection state in a couple of runs. Better to just have it here to keep from repeating myself.

Reply:
You're correct after all. I'm not updating the array. I'm now just checking the correct auth method is called.